python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,397 @@
+
+    Parameters
+    ----------
+    domain : `lsst.ts.salobj.Domain`
+        Domain for remotes. If `None` create a domain.
+    log : `logging.Logger`
+        Optional logging class to be used for logging operations. If `None`,
+        creates a new logger. Useful to use in salobj.BaseScript and allow
+        logging in the class use the script logging.
+    intended_usage: `int`
+        Optional integer that maps to a list of intended operations. This is
+        used to limit the resources allocated by the class by gathering some
+        knowledge about the usage intention. By default allocates all
+        resources.
+    latiss: LATISS
+    """
+
+    def __init__(
+        self,
+        domain: typing.Optional[salobj.Domain] = None,
+        log: typing.Optional[logging.Logger] = None,
+        intended_usage: typing.Optional[int] = None,
+        latiss: typing.Optional[LATISS] = None,
+    ) -> None:
+
+        self.electrometer_index = 1
+        self.fiberspectrograph_index = 1
+
+        super().__init__(
+            components=[
+                "ATMonochromator",
+                "ATWhiteLight",
+                f"FiberSpectrograph:{self.fiberspectrograph_index}",
+                f"Electrometer:{self.electrometer_index}",
+            ],
+            instrument_setup_attributes=[
+                "filter",
+                "grating",
+                "entrance_slit",
+                "exit_slit",
+            ],
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+        )
+
+        self.latiss = latiss
+
+        self.entrance_slit = 0.5
+        self.exit_slit = 0.5
+
+        self.whitelight_startup_power = 1000
+
+        self.chiller_temp_tolerance_relative = 0.2
+        self.chiller_temperature = 20
+
+        self.exptime_dict: dict[str, float] = dict(
+            camera=0.0,
+            electrometer=0.0,
+            fiberspectrograph=0.0,
+        )
+        self.delay = 1
+
+    async def change_wavelength(self, wavelength: float) -> None:
+        """Change the ATMonochromator wavelength setting"""
+
+        await self.rem.atmonochromator.cmd_changeWavelength.set_start(
+            wavelength=wavelength, timeout=self.long_long_timeout
+        )
+
+    async def is_ready_for_flats(self) -> bool:
+        """Add doctring"""
+        # TODO (DM-44310): Implement method to check that the
+        # system is ready for flats.
+        return True
+
+    async def setup_calsys(self, **kwargs: typing.Union[int, float, str, None]) -> None:
+        """Turn on the Chiller and the ATWhiteLight and wait for the
+        warmup to complete
+        """
+        await self.start_chiller()
+        await self.wait_for_chiller_temp_within_tolerance()
+        await self.switch_lamp_on()
+        await self.wait_for_lamp_to_warm_up()
+
+    async def configure_flat(
+        self,
+        wavelength: typing.Union[float, None],
```

Comment:
What you are trying to accomplish by allowing the value of the `wavelength` argument to be `None`?

The way it is right now, the user would have to call this method like:

```
await atcalsys.configure_flat(wavelength = None)
```

to have it fallback to using the "default" value of 500. I think you might be trying to provide a simple way that users could call `configure_flat` with no input at all. If that is the case I suggest you add a default value for `wavelength`. Something like:

```
async def configure_flat(self, wavelength: float = 500.0, ...) -> None:
``` 

This would allow users to do something like:

```
await atcalsys.configure_flat()
```
e.g. calling it with no argument would fallback to using 500 (or whatever value is the default).

However, I am not sure this is a good idea. You might want to stick with the original interface (at least when it comes to wavelength) and force users to always pass a value. 

This means the method definition would be something like

```
async def configure_flat(self, wavelength: float, ...) -> None:
```

And users would have to pass in a wavelength value of type float. 

However, looking at how you are later relying on `kwargs` to get additional input values, I think we may have to do some work to improve this interface in the future. I think it is fine for now to use it like this, but something to keep in mind for future improvements.

Reply:
now done with configuration files